gcp-auth-extension/src/main/java/io/opentelemetry/contrib/gcp/auth/GcpAuthAutoConfigurationCustomizerProvider.java

```diff
@@ -228,10 +234,36 @@ private static Map<String, String> getRequiredHeaderMap(
   }
 
   // Updates the current resource with the attributes required for ingesting OTLP data on GCP.
-  private static Resource customizeResource(Resource resource, ConfigProperties configProperties) {
-    String gcpProjectId =
-        ConfigurableOption.GOOGLE_CLOUD_PROJECT.getConfiguredValue(configProperties);
-    Resource res = Resource.create(Attributes.of(stringKey(GCP_USER_PROJECT_ID_KEY), gcpProjectId));
+  private static Resource customizeResource(Resource resource, String gcpProjectId) {
+    Resource res =
+        Resource.create(
+            Attributes.of(AttributeKey.stringKey(GCP_USER_PROJECT_ID_KEY), gcpProjectId));
     return resource.merge(res);
   }
+
+  /**
+   * Retrieves the Google Cloud Project ID from the configuration properties, falling back to
+   * google-cloud-core's ServiceOptions project ID resolution if not explicitly set.
+   *
+   * @param configProperties The configuration properties containing the GCP project ID.
+   * @return The Google Cloud Project ID.
+   */
+  @Nonnull
+  static String getGoogleProjectId(
```

Comment:
We could do away with this functions since much of this functionality is already available in `ConfigurableOption` class.

See suggestion for `customizeResource` method.

Reply:
Hi @gustavovnicius, this method is an implementation detail of the class and should have been private. For testing, I think you should test the expected behavior instead of testing the method in isolation. I think `testQuotaProjectBehavior` does something similar. 

 So an idea here would be to mock the `GoogleCredentials` (particularly the `getProjectId` method) and in one of the cases, you return a valid Project ID and see if the extension loads or throws an expected `GoogleAuthException`

Let me know if this makes sense.